### PR TITLE
[SPARK-8676][SQL]Lazy start event logger in sql application to avoid TGT expiring in l…

### PR DESCRIPTION
**Pull request by WangTaoTheTonic**
…ong connection

Now in Thrift Server/Spark SQL, it will login first in `Client.scala`, then open the file stream in event logger. After that the login action will be executed twice again, one of them is in `SparkSQLCLIService`, and another is in `ThriftHttpCLIService/ThriftBinaryCLIService`.

If we open a long-connected file stream to HDFS between two login, the UserGroupInformation that RPC took will be refreshed by second login. After the TGT expires, it will cause "No invalid credentials" excepiton.

So in SQL application, we will start event logger only after the HiveServer2 being launched.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala [hunk 1]
```diff
@@ -185,9 +194,23 @@ private[spark] class EventLoggingListener(
...
 
   override def onApplicationStart(event: SparkListenerApplicationStart): Unit = {
+    if (!sparkConf.get("spark.app.name").startsWith("SparkSQL::")) {
+      start()
+      started.compareAndSet(false, true)
+    }
     logEvent(event, flushLogger = true)
   }
 
+  def onSQLApplicationStart(): Unit = {
```
Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala]: we shouldn't have SQL specific things in core

**Comment by andrewor14** [comment]
@vanzin @harishreedharan can you have a look? This seems like a YARN issue according to the issue description.

**Comment by andrewor14** [comment]
This patch has gone stale; we now have a `SQLListener` that may be relevant here. I would recommend that we close this patch and re-open it later against the latest changes in master if there is interest.

**Comment by WangTaoTheTonic** [comment]
No prob. will close this.

**Pull request closed without merging**